monitoring/scd_qualifier/utils.py

```diff
@@ -0,0 +1,117 @@
+
+class VolumePolygon(ImplicitDict):
+    ''' A class to hold the polygon object, used in the outline_polygon of the Volume3D object as specified in the ASTM standard '''
+    vertices: List[LatLngPoint] # A minimum of three LatLngPoints
+
+class Circle(ImplicitDict):
+    ''' Hold the details of a circle object used in the outline_circle object as specified in the ASTM standard '''
+    center: LatLngPoint 
+    radius: Radius
+
+
+class Altitude(ImplicitDict):
+    ''' A class to hold altitude per ASTM standard '''
+    value:int
+    reference:str 
+    units: str 
+
+class Time(ImplicitDict):
+    ''' A class to hold Time per ASTM standard'''
+    value: str
+    format:str 
+
+class OperationalIntentReference(ImplicitDict):
+    """Class for keeping track of an operational intent reference"""
+    id: str
+
+class Volume3D(ImplicitDict):
+    '''A class to hold Volume3D objects '''
+    
+    outline_polygon: VolumePolygon
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+
+class Volume4D(ImplicitDict):
+    '''A class to hold ASTM Volume4D objects '''
+    volume: Volume3D
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+
+
+class OperationalIntentDetails(ImplicitDict):
+    """Class for holding details of an operational intent reference """
+    volumes: List[Volume4D]
+    priority: int
+
+class VolumeGenerationRule(ImplicitDict):
+    """ A class to hold configuration for holding rule for volume generation """
+    intersect_altitude: bool = 0
+    intersect_time: bool = 0
+    is_control: bool = 0
+    expected_result: str
+
+class GeoJSONFeature(ImplicitDict):
+    ''' A class to hold GeoJSON Feature flight plan, at this moment we are using a LineString but it could be a LineString or Polygon '''
+    geojson: Feature
+    
+class SCDVolume4D(ImplicitDict):
+    ''' A class to hold a volume 4D with path and time details and GeoJSON flight plan, USSP will have to transform flight plan to a  '''
+    flight_plan: GeoJSONFeature
+    time_start: StringBasedDateTime
+    time_end: StringBasedDateTime
+    altitude_lower: Altitude
+    altitude_upper: Altitude
+    
+class GeometryGenerationRule(ImplicitDict):
+    """ A class to hold configuration for developing treatment flight path """
+    intersect_space:bool = 0
+    
+class GeometryPayload(ImplicitDict):
+    ''' An object to hold generated flight path, is_control is a nomenclature used to see if the generated path is the first one '''
+    geometry: Union[LineString, Polygon]
+    is_control: bool
+    geometry_generation_rule: GeometryGenerationRule
+  
+class FlightAuthPayload(ImplicitDict):
+    ''' An object to hold flight authorization details '''
+    uas_serial_number:str
+    operation_mode: str
+    operation_category:str
+    uas_class:str
+    identification_technologies:List =[]
+    connectivity_methods:List = []
+    endurance_minutes: int 
+    emergency_procedure_url: str
+    operator_id:str
+
+class SCDTestPayload(ImplicitDict):
```

Comment:
This seems useful but doesn't seem to be used anywhere and doesn't appear in the API, so it doesn't seem like we should add it yet.

Reply:
OK I will remove it. We can add it when we combine the test 1 (operator data etc. that the FlightAuthPayload) carries in to this. My idea is that this structure is what should be sent to the USS who then processes it.